src/zenml/integrations/kubernetes/orchestrators/kubernetes_orchestrator_entrypoint.py

```diff
@@ -505,6 +505,67 @@ def start_step_job(node: Node) -> NodeStatus:
 
             return NodeStatus.RUNNING
 
+        def stop_step(node: Node) -> None:
+            """Stop a step.
+
+            Args:
+                node: The node to stop.
+            """
+            step_name = node.id
+
+            label_selector = (
+                f"run_id={kube_utils.sanitize_label(str(pipeline_run.id))},"
+                f"step_name={kube_utils.sanitize_label(node.id)}"
```

Comment:
After this sanitization of the step name, they might not be unique anymore, which is why this might find multiple steps. I've introduced annotations with the step name that should be used to verify this